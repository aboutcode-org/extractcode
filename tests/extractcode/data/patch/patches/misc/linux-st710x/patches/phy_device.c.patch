--- linux-2.6.23/drivers/net/phy/phy_device.c.orig	2009-02-04 16:33:05.000000000 +0100
+++ linux-2.6.23/drivers/net/phy/phy_device.c	2009-02-04 16:58:56.000000000 +0100
@@ -161,10 +161,20 @@
 
 	phy_prepare_link(phydev, handler);
 
+	/* Make sure higher levels (i.e. Ethtool) returns link
+	   down before the PHY has been properly initialized. */
+	netif_carrier_off(phydev->attached_dev);
+
 	phy_start_machine(phydev, NULL);
 
 	if (phydev->irq > 0)
 		phy_start_interrupts(phydev);
+	else
+		/* Make sure phy interrupts are used. Please verify
+		   that a correct interrupt number is specified in
+		   the setup functions of the board if this BUG has
+		   been triggered. */
+		BUG();
 
 	return phydev;
 }
@@ -671,8 +681,13 @@
 	phydev->drv = phydrv;
 
 	/* Disable the interrupt if the PHY doesn't support it */
-	if (!(phydrv->flags & PHY_HAS_INTERRUPT))
+	if (!(phydrv->flags & PHY_HAS_INTERRUPT)) {
+		/* Make sure phy interrupts are used. Please verify
+		   that the phy is registered before the MAC if this
+		   BUG has been triggered. */
+		BUG();
 		phydev->irq = PHY_POLL;
+	}
 
 	spin_lock_bh(&phydev->lock);
 
