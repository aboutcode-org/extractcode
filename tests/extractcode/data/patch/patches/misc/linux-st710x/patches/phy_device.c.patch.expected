[
  [
    "linux-2.6.23/drivers/net/phy/phy_device.c.orig", 
    "linux-2.6.23/drivers/net/phy/phy_device.c", 
    [
      "--- linux-2.6.23/drivers/net/phy/phy_device.c.orig", 
      "+++ linux-2.6.23/drivers/net/phy/phy_device.c", 
      "@@ -161,10 +161,20 @@", 
      "", 
      "phy_prepare_link(phydev, handler);", 
      "", 
      "+\t/* Make sure higher levels (i.e. Ethtool) returns link", 
      "+\t   down before the PHY has been properly initialized. */", 
      "+\tnetif_carrier_off(phydev->attached_dev);", 
      "+", 
      "phy_start_machine(phydev, NULL);", 
      "", 
      "if (phydev->irq > 0)", 
      "phy_start_interrupts(phydev);", 
      "+\telse", 
      "+\t\t/* Make sure phy interrupts are used. Please verify", 
      "+\t\t   that a correct interrupt number is specified in", 
      "+\t\t   the setup functions of the board if this BUG has", 
      "+\t\t   been triggered. */", 
      "+\t\tBUG();", 
      "", 
      "return phydev;", 
      "}", 
      "@@ -671,8 +681,13 @@", 
      "phydev->drv = phydrv;", 
      "", 
      "/* Disable the interrupt if the PHY doesn't support it */", 
      "-\tif (!(phydrv->flags & PHY_HAS_INTERRUPT))", 
      "+\tif (!(phydrv->flags & PHY_HAS_INTERRUPT)) {", 
      "+\t\t/* Make sure phy interrupts are used. Please verify", 
      "+\t\t   that the phy is registered before the MAC if this", 
      "+\t\t   BUG has been triggered. */", 
      "+\t\tBUG();", 
      "phydev->irq = PHY_POLL;", 
      "+\t}", 
      "", 
      "spin_lock_bh(&phydev->lock);", 
      ""
    ]
  ]
]